### PR TITLE
update comment with EnqueueExtensions

### PR DESCRIPTION
**Pull request by kerthcet**
Signed-off-by: kerthcet <kerthcet@gmail.com>

#### What type of PR is this?
/sig scheduling
/kind documentation

#### What this PR does / why we need it:
add comment for EnqueueExtensions
#### Which issue(s) this PR refs:

Refs #103633

#### Special notes for your reviewer:
New to community,  hope to hear as much advices as better.
#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/scheduler/framework/interface.go [hunk 1]
```diff
@@ -307,10 +307,17 @@ type QueueSortPlugin interface {
...
+// plugins may impacting scheduling, other implemented plugins are useless and
+// harmless.
+// When setting up a new scheduler, it will load events registered by
+// EventsToRegister(), and convert them to gvks to build dynamic event handlers
+// with.
 type EnqueueExtensions interface {
-	// EventsToRegister returns a series of interested events that
-	// will be registered when instantiating the internal scheduling queue.
+	// EventsToRegister returns a series of possible events that may cause a Pod
+	// scheduling failed by this plugin.
```
Comment (Huang-Wei) [inline pkg/scheduler/framework/interface.go:318]:
```suggestion
	// failed by this plugin schedulable.
```

File: pkg/scheduler/framework/interface.go [hunk 2]
```diff
@@ -307,10 +307,17 @@ type QueueSortPlugin interface {
...
+	// EventsToRegister() is called to fill ClusterEvents to frameworkOptions's
+	// clusterEventMap, which will be used to build event handlers.
```
Comment (Huang-Wei) [inline pkg/scheduler/framework/interface.go:320]:
```
// The events will be registered when instantiating the internal scheduling queue,
// and leveraged to build event handlers dynamically.
```

**Comment by kerthcet** [comment]
hey @Huang-Wei , thanks for your advices. would you mind to review the updated comments again.

File: pkg/scheduler/framework/interface.go [hunk 3]
```diff
@@ -307,10 +307,16 @@ type QueueSortPlugin interface {
 }
 
 // EnqueueExtensions is an optional interface that plugins can implement to efficiently
-// move unschedulable Pods in internal scheduling queues.
+// move unschedulable Pods in internal scheduling queues. Only plugins
+// impacting scheduling will take effect, other implemented plugins are useless and
+// harmless.
+// When setting up a new scheduler, it will load events and convert them to gvks
+// to build dynamic event handlers with.
```
Comment (chendave) [inline pkg/scheduler/framework/interface.go:314]:
```suggestion
// to dynamically build event handlers.
```
or simply remove this line, as it is a redundant information.

File: pkg/scheduler/framework/interface.go [hunk 4]
```diff
@@ -307,10 +307,16 @@ type QueueSortPlugin interface {
 }
 
 // EnqueueExtensions is an optional interface that plugins can implement to efficiently
-// move unschedulable Pods in internal scheduling queues.
+// move unschedulable Pods in internal scheduling queues. Only plugins
+// impacting scheduling will take effect, other implemented plugins are useless and
+// harmless.
```
Comment (chendave) [inline pkg/scheduler/framework/interface.go:312]: not quite clear to me, do we have any plugin doesn't impact scheduling at all? also, I cannot understand this statement 
> other implemented plugins are useless and harmless.

Comment (kerthcet) [reply pkg/scheduler/framework/interface.go]: sorry about the ambiguous words，I updated my comments right now. 🥲

File: pkg/scheduler/framework/interface.go [hunk 5]
```diff
@@ -307,10 +307,14 @@ type QueueSortPlugin interface {
 }
 
 // EnqueueExtensions is an optional interface that plugins can implement to efficiently
-// move unschedulable Pods in internal scheduling queues.
+// move unschedulable Pods in internal scheduling queues. Plugins
+// who will fail pod scheduling should implement this interface.
```
Comment (Huang-Wei) [inline pkg/scheduler/framework/interface.go:311]:
```suggestion
// that fail pod scheduling (e.g., Filter plugins) are expected to implement this interface.
```

File: pkg/scheduler/framework/interface.go [hunk 6]
```diff
@@ -307,10 +307,14 @@ type QueueSortPlugin interface {
...
+// With registered events, scheduler can build dynamically event handlers.
```
Comment (Huang-Wei) [inline pkg/scheduler/framework/interface.go:312]: It's described in L316~L317, so let's remove it.

**Review by Huang-Wei — COMMENTED** [review]
Some nits. LGTM otherwise.

File: pkg/scheduler/framework/interface.go (hunk 6 above)
Comment (kerthcet) [reply pkg/scheduler/framework/interface.go]: yep, it's duplicated👍

**Comment by kerthcet** [comment]
Updated the comments as advised. Thanks for all your patience @Huang-Wei @chendave , it's my first pr, twist but happy. Looking forward to contributing more codes. 💪

**Pull request merged**